x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java

```diff
@@ -419,9 +505,55 @@ private static void unaryNumeric(
                         () -> BigInteger.valueOf(Long.MAX_VALUE).add(BigInteger.valueOf(randomLongBetween(1, Integer.MAX_VALUE)))
                     )
                 )
-            )
+            ),
+            Map.entry(DataTypes.BOOLEAN, List.of(Map.entry("<true>", () -> true), Map.entry("<false>", () -> false))),
+            Map.entry(
+                DataTypes.DATETIME,
+                List.of(
+                    Map.entry("<1970-01-01T00:00:00Z>", () -> 0L),
+                    Map.entry(
+                        "<date>",
+                        () -> randomLongBetween(0, 10 * (long) 10e11) // 1970-01-01T00:00:00Z - 2286-11-20T17:46:40Z
+                    ),
+                    Map.entry(
+                        "<far future date>",
+                        () -> randomLongBetween(10 * (long) 10e11, Long.MAX_VALUE) // 2286-11-20T17:46:40Z - +292278994-08-17T07:12:55.807Z
+                    )
+                )
+            ),
+            Map.entry(
+                DataTypes.IP,
+                List.of(
+                    Map.entry("<ipv4>", () -> new BytesRef(InetAddressPoint.encode(randomIp(true)))),
+                    Map.entry("<ipv6>", () -> new BytesRef(InetAddressPoint.encode(randomIp(false))))
+                )
+            ),
+            Map.entry(DataTypes.KEYWORD, stringRandomValues("keyword")),
+            Map.entry(DataTypes.TEXT, stringRandomValues("text")),
+            Map.entry(DataTypes.VERSION, versionRandomValues(""))
         );
```

Comment:
thought (massively out-of-scope for this PR, though): the fact that this is merely a map, even though we know all data types at compile time, makes me think that things could be easier for us if we could use enums rather than maps (easier refactoring, compile-time validation of many parts of the code, auto-completion in Intellij etc.). I realize that this is not possible due to the fact that `DataTypes` is a collection of constants, rather than an enum, and it also sits in the `ql` package, which we probably cannot easily refactor without affecting lots of other code.